From eded421b8bd35a160e4cad537b2227fece61c03d Mon Sep 17 00:00:00 2001
From: David McKay <david.mckay@st.com>
Date: Thu, 28 May 2009 11:58:27 +0100
Subject: [PATCH] sh_stm: Add more clock divisors on 7105,7111,7141

Adds /12 and /24 ratios to the clock code for the 7105, 7111, and
7141. This is needed primarily for PCI to get the 33MHz and 66MHz clocks
which are the most common PCI clocks used. Really this code needs to be
rewritten to support the hardware properly as the divisors are now
arbitrary integers instead of a restricted set as they were on some
earlier parts.

Signed-off-by: David McKay <david.mckay@st.com>
---
 arch/sh/kernel/cpu/sh4/clock-stx7105.c |    2 ++
 arch/sh/kernel/cpu/sh4/clock-stx7111.c |    2 ++
 arch/sh/kernel/cpu/sh4/clock-stx7141.c |    2 ++
 3 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/clock-stx7105.c b/arch/sh/kernel/cpu/sh4/clock-stx7105.c
index 257e98c..690fe13 100644
--- a/arch/sh/kernel/cpu/sh4/clock-stx7105.c
+++ b/arch/sh/kernel/cpu/sh4/clock-stx7105.c
@@ -285,7 +285,9 @@ static const struct xratio ratios [] = {{1, 0x10000 },
                                         {2, 0x1     },
                                         {4, 0x3     },
                                         {8, 0x7     },
+					{12, 0xb    },
                                         {16, 0xf    },
+					{24, 0x17   },
                                         {32, 0x1f   },
                                         {NO_MORE_RATIO, }
 };
diff --git a/arch/sh/kernel/cpu/sh4/clock-stx7111.c b/arch/sh/kernel/cpu/sh4/clock-stx7111.c
index b68e844..f7d1429 100644
--- a/arch/sh/kernel/cpu/sh4/clock-stx7111.c
+++ b/arch/sh/kernel/cpu/sh4/clock-stx7111.c
@@ -272,7 +272,9 @@ static const struct xratio ratios [] = {{1,  0x10000 },
 					{2,  0x1 },
 					{4,  0x3 },
 					{8,  0x7 },
+					{12, 0xb },
 					{16, 0xf },
+					{24, 0x17 },
 					{32, 0x1f },
 					{NO_MORE_RATIO, }
 };
diff --git a/arch/sh/kernel/cpu/sh4/clock-stx7141.c b/arch/sh/kernel/cpu/sh4/clock-stx7141.c
index ff35db8..92f98f0 100644
--- a/arch/sh/kernel/cpu/sh4/clock-stx7141.c
+++ b/arch/sh/kernel/cpu/sh4/clock-stx7141.c
@@ -244,7 +244,9 @@ static const struct xratio ratios [] = {{1,  0x0 },
 					{2,  0x1 },
 					{4,  0x3 },
 					{8,  0x7 },
+					{12, 0xb },
 					{16, 0xf },
+					{24, 0x17 },
 					{32, 0x1f },
 					{NO_MORE_RATIO, }
 };
-- 
1.6.0.6

